xcm-simulator/src/tests/laosish_xcm.rs

```diff
@@ -143,40 +144,73 @@ fn xcmp_create_foreign_asset() {
 	});
 }
 
-#[ignore] // TODO
 #[test]
-fn xcmp_teleport_native_assets_to_asset_hub() {
+fn roundtrip_teleport_laosish_to_assethub() {
 	MockNet::reset();
 
-	let para_a_native_asset_location =
-		xcm::v3::Location::new(1, [xcm::v3::Junction::Parachain(PARA_LAOSISH_ID)]);
+	let laosish_native_asset_location = Location::new(1, [Junction::Parachain(PARA_LAOSISH_ID)]);
 
-	let create_asset = parachain::RuntimeCall::ForeignAssets(ForeignAssetsCall::create {
-		id: para_a_native_asset_location,
+	let create_asset = asset_hub::RuntimeCall::ForeignAssets(AssetHubAssetsCall::create {
+		id: laosish_native_asset_location,
 		admin: sibling_account_id(PARA_LAOSISH_ID),
-		min_balance: 1000,
+		min_balance: 1,
 	});
 
+	let teleport_amount_1 = 100;
+	let teleport_amount_2 = 10;
+
+	let alith: laosish::AccountId = ALITH.into();
+
 	Laosish::execute_with(|| {
 		assert_ok!(LaosishPalletXcm::send_xcm(
 			Here,
-			(Parent, Parachain(laosish::configs::xcm_config::ASSET_HUB_ID)),
+			(Parent, Parachain(PARA_ASSETHUB_ID)),
 			Xcm(vec![Transact {
 				origin_kind: OriginKind::Xcm,
 				require_weight_at_most: Weight::from_parts(INITIAL_BALANCE as u64, 1024 * 1024),
 				call: create_asset.encode().into(),
 			}]),
 		));
+
+		assert_eq!(laosish::Balances::free_balance(alith), INITIAL_BALANCE);
+
+		assert_ok!(LaosishPalletXcm::limited_teleport_assets(
+			laosish::RuntimeOrigin::signed(ALITH.into()),
+			Box::new((Parent, Parachain(PARA_ASSETHUB_ID)).into()),
+			Box::new(AccountId32 { network: None, id: ALICE.into() }.into()),
```

Comment:
overall good test because you proved that it's possible to teleport assets in both directions.
However it does not prove our main use case :
- Alith teleport to assethub and back

Alith does not have Alice privatekey. Alith has only Alith private key

Reply:
Resolved offline: this is not possible, as AssetHub has 32-bytes addresses